### PR TITLE
Moka with plugin architecture

### PR DESCRIPTION
**Pull request by Giuffre**
With this first implementation we allowing Moka  to be extended by any user defined plugin.

### CODE REVIEW

File: src/Moka/Plugin/PluginHelper.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    /**
+     * The base namespace with plugin must be declared
+     */
+    const PLUGIN_NAMESPACE_TEMPLATE = 'Moka\\Plugin\\%s\\%sPlugin';
+
+    /**
+     * @param string $pluginName
+     * @return string
+     */
+    public static function generatePluginFQCN(string $pluginName): string
```
Comment (xzhayon) [inline src/Moka/Plugin/PluginHelper.php]: Does it need to be public?

File: src/Moka/Plugin/PHPUnit/PHPUnitPlugin.php [hunk 2]
```diff
@@ -6,11 +6,19 @@
 use Moka\Exception\PluginNotRegisteredException;
 use Moka\Moka;
 use Moka\Plugin\PluginInterface;
+use Moka\Strategy\MockingStrategyInterface;
```
Comment (xzhayon) [inline src/Moka/Plugin/PHPUnit/PHPUnitPlugin.php]: The circular dependency is still there as plugins require interfaces defined in Moka.

**Review by xzhayon — CHANGES_REQUESTED** [review]
Nice implementation, but still some cleanup required imho.

File: src/Moka/Plugin/PHPUnit/PHPUnitPlugin.php (hunk 2 above)
Comment (xzhayon) [reply src/Moka/Plugin/PHPUnit/PHPUnitPlugin.php]: Actually, this is fine, but it means we still need to consider a meta-package.

File: src/Moka/Plugin/PluginHelper.php (hunk 1 above)
Comment (Giuffre) [reply src/Moka/Plugin/PluginHelper.php]: Done. Make as private

File: tests/Strategy/PHPUnitMockingStrategyTest.php [hunk 3]
```diff
@@ -14,7 +13,7 @@ public function __construct($name = null, array $data = [], $dataName = '')
     {
         parent::__construct($name, $data, $dataName);
 
-        $this->setStrategy(new PHPUnitMockingStrategy());
+        $this->setStrategy(new \Moka\Plugin\PHPUnit\PHPUnitMockingStrategy());
```
Comment (xzhayon) [inline tests/Strategy/PHPUnitMockingStrategyTest.php]: Whoops!

**Review by xzhayon — CHANGES_REQUESTED** [review]
OK for me! Just a small issue...

**Review by xzhayon — APPROVED** [review]

**Review by xzhayon — APPROVED** [review]
All fine for me!

**Pull request merged**